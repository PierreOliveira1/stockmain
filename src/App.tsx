import React, { useState } from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';

const App = () => {
	const [count, setCount] = useState(0);

	const addCount = () => {
		setCount(count + 1);
	};

	return (
		<View style={styles.container}>
			<Text>Hello World!!!</Text>
			<Text testID="count">{count}</Text>
			<TouchableOpacity testID="buttonAddCount" onPress={addCount}>
				<Text>Add</Text>
			</TouchableOpacity>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center',
		flexDirection: 'column',
	},
});

export default App;
