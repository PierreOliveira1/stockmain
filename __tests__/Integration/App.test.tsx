import { render, fireEvent } from '@testing-library/react-native';
import App from '../../src/App';

describe('App', () => {
	test('should get hello world in app', () => {
		const { getByText } = render(<App />);
		const textElement = getByText(/hello world/i);
		expect(textElement).toBeTruthy();
	});

	test('should get count 0 in app', () => {
		const { getByTestId, toJSON } = render(<App />);
		const count = getByTestId('count');
		expect(count.props.children).toBe(0);
		expect(toJSON()).toMatchSnapshot();
	});

	test('should get count 1 in app', () => {
		const { getByTestId, toJSON } = render(<App />);
		const buttonAdd = getByTestId('buttonAddCount');
		fireEvent.press(buttonAdd);
		const count = getByTestId('count');
		expect(count.props.children).toBe(1);
		expect(toJSON()).toMatchSnapshot();
	});
});
