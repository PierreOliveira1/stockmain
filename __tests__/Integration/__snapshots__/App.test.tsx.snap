// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should get count 0 in app 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello World!!!
  </Text>
  <Text
    testID="count"
  >
    0
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="buttonAddCount"
  >
    <Text>
      Add
    </Text>
  </View>
</View>
`;

exports[`App should get count 1 in app 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello World!!!
  </Text>
  <Text
    testID="count"
  >
    1
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="buttonAddCount"
  >
    <Text>
      Add
    </Text>
  </View>
</View>
`;
